bukuserver/views.py

```diff
@@ -195,13 +195,20 @@ def __init__(self, bukudb: buku.BukuDb, *args, **kwargs):
         custom_model = types.SimpleNamespace(bukudb=bukudb, __name__='bookmark')
         super().__init__(custom_model, *args, **kwargs)
 
+    @property
+    def url_render_mode(self):
+        return app_param('URL_RENDER_MODE', DEFAULT_URL_RENDER_MODE)
+
     @property
     def page_size(self):
         return app_param('PER_PAGE', DEFAULT_PER_PAGE)
 
     @property
-    def url_render_mode(self):
-        return app_param('URL_RENDER_MODE', DEFAULT_URL_RENDER_MODE)
+    def page_size_options(self):
+        return tuple(sorted(set([self.page_size] + list(super().page_size_options))))
+
+    def get_safe_page_size(self, page_size):  # un-enforcing the restriction
+        return (page_size if self.can_set_page_size and page_size > 0 else self.page_size)
```

Comment:
Allowing for custom page sizes (which get rejected by the [default behaviour](https://github.com/pallets-eco/flask-admin/pull/2479/files#diff-81459308eab181f696582f7b64e36fb5616b7106eeeae2d2284535d632106789R1521-R1528))